### ISSUE TITLE
Rename posterior class

### ISSUE THREAD

**Issue by MansMeg** [MansMeg/posteriordb#52]
In posteriordb, how do we name the class containing a posterior? Simply "posterior" right? Perhaps we could think of a more specific class name as I fear this could crash with other packages' class names

**Comment by MansMeg** [comment]
@paul-buerkner and @eerolinna do you have any suggestions?

**Comment by paul-buerkner** [comment]
I would argue for a more specific class name indeed. We could have a simple `db_` or `pdb_` in front of the object name to avoid any name clashing I would say.

**Comment by MansMeg** [comment]
That make sense to me. Any thoughts on that @eerolinna ?

**Comment by eerolinna** [comment]
I'm fine with having a prefix in R. In python it won't be necessary because each package has a separate namespace

**Comment by MansMeg** [comment]
Its the same in R so it is not a namespace problem, more that the Stan community is growing and I think posterior will be to generic in this ecosystem. Hence making it more specific is a long-term decision. I guess that is your though @paul? 

If this is the cade I guess we should do the same both in R and Python (at least in the long run).

**Comment by paul-buerkner** [comment]
Yes this is my point.

Måns Magnusson <notifications@github.com> schrieb am Mi., 16. Okt. 2019,
18:15:

**Comment by eerolinna** [comment]
So you mean there is a conceptual difference between the term posterior and the thing we currently call posteriors? So that the pdb_posterior (or whatever we end up calling it) doesn't mean the same thing as what posterior generally means?

**Comment by eerolinna** [comment]
Just to check I understand R correctly, would this not result in a namespace clash?

```R
library(posteriordb) # this exports a function called posterior
library(otherlib) # this also exports a function called posterior
po <- posterior("8_schools-8_schools_centered", ...)
```

**Comment by paul-buerkner** [comment]
It may not be misleading but the class name "posterior" is very general and so might be used by other packages so that methods may end up clashing. So it is not only about functions (they would be masking one another) but about the same object names. There is no drawback in naming the object pdb_posterior as the regular user never needs to see these names anyway. So we should go for something that wont produce problems along the way.

**Comment by eerolinna** [comment]
Ah so we are talking about renaming the internal class name (which by itself doesn't affect the user facing API), so essentially `posterior` to `pdb_posterior` here?
https://github.com/MansMeg/posteriordb/blob/cc83694ec8d024c0a0faa80d916c595444e78802/rpackage/R/posterior.R#L24

That sounds good.

Would
```R
po <- posterior("8_schools-8_schools_centered", my_pdb)
```
remain as it is now or should it also be renamed to `pdb_posterior`?

For python I still don't think a prefix is necessary. Perhaps an example illustrates why
```python
import posteriordb # this exports a class called posterior
import otherlib # this also exports a class called posterior

# no clash here
po1 = posteriordb.posterior(...) # object of class posteriordb.posterior
po2 = otherlib.posterior(...) # object of class otherlib.posterior

# no clash here either
po1.print() # this could be any method
po2.print()
```
There's no clashes as essentially the name of the package is automatically used as a prefix

**Comment by paul-buerkner** [comment]
The function name would stay the same.

**Comment by paul-buerkner** [comment]
I would say yes.

**Comment by MansMeg** [comment]
Great! Then Ill fix this now.

**Comment by MansMeg** [comment]
Alright. Now, this is done. If you have time you could check if the current code is as you expected. If you are happy with it you can close this issue @paul-buerkner .

**Comment by paul-buerkner** [comment]
Class names look good for me. The only question I have is whether we really do need all the aliases, for instance, `pdb_posterior` for `posterior`. I see the reason behind this (the class is named pdb_posterior) but in the R world nobody really expects the constructors to be named exactly as the class. That is, what is important is the class name itself but less so the function name I would say. Function masking is usually explicit and predictable but method dispatch may potentially be weird if two different objects have the same classe name. 

Long story short, we may remove the function name aliases without causing harm but we can also keep them just to consistency. Both is fine for me.

**Comment by MansMeg** [comment]
Great! You know me. I really like consistency. =)
